### ISSUE TITLE
"referenceName" requestParameters behaves in an unexpected way

### ISSUE THREAD

**Issue by albodrug** [EGA-archive/beacon2-ri-api#388]
Hello, 

When sending a request to the Beacon API with "referenceName" parameters, it returns all variants found in chromosomes containing the number/string passed tp "referenceName". So if I ask for "referenceName": 1, i have variants appearing in chromosome 1 and 19. Same when asking for "referenceName":2 I get variants appearing in chromosome 2 and 22.

(just to be precise in my dataset I have chromosomes 1 to 6, 19 and 22)

See below:

`> curl -X POST http://localhost:5050/api/g_variants -H "Content-Type: application/json" -d '{
  "meta": {
    "apiVersion": "2.0"
  },
  "query": {
    "requestParameters": {
        "referenceName": "1"
    },
    "filters": [],
    "includeResultsetResponses": "HIT",
    "pagination": {
      "skip": 0,
      "limit": 1000
    },
    "testMode": false,
    "requestedGranularity": "record"
  }
}' |python -m json.tool | grep genomicHGVSId | sed 's,:g.*,,' | sort | uniq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  680k    0  679k  100   307   407k    184  0:00:01  0:00:01 --:--:--  408k
                            "genomicHGVSId": "NC_00001.11
                            "genomicHGVSId": "NC_000019.11
`
`> curl -X POST http://localhost:5050/api/g_variants -H "Content-Type: application/json" -d '{
  "meta": {
    "apiVersion": "2.0"
  },
  "query": {
    "requestParameters": {
        "referenceName": "2"
    },
    "filters": [],
    "includeResultsetResponses": "HIT",
    "pagination": {
      "skip": 0,
      "limit": 1000
    },
    "testMode": false,
    "requestedGranularity": "record"
  }
}' |python -m json.tool | grep genomicHGVSId | sed 's,:g.*,,' | sort | uniq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  420k    0  420k  100   307   334k    244  0:00:01  0:00:01 --:--:--  334k
                            "genomicHGVSId": "NC_00002.11
                            "genomicHGVSId": "NC_000022.11`
                            
I'd think this is not a desirable behaviour?

I am using the official release from March, maybe you fixed it since?

Thanks!
Alex

**Comment by costero-e** [comment]
Hi @albodrug,

Thank you! I have fixed this in the last PR. Can you update your beacon and try again?

Best,

Oriol

**Comment by albodrug** [comment]
Hello, 

The POST requests and the GET filters don't seem to work with the last pull?
`(master)> git log -1
commit f5f3454fc90b7f09c4f1f486bf7e9af528a63083 (HEAD -> master, origin/master, origin/HEAD)
`
simple GET works
`> http GET http://localhost:5050/api/g_variants
HTTP/1.1 200 OK
Content-Type: application/json;charset=utf-8
Date: Mon, 18 Nov 2024 15:24:38 GMT
Server: Beacon Reference Implementation demo v2.0-842c290 (based on Python/3.10 aiohttp/3.8.1)
Transfer-Encoding: chunked
{
    "beaconHandovers": [
        {
            "handoverType": {
                "id": "CUSTOM:000001",
                "label": "Project description"
            },
            "note": "Project description",
            "url": "https://www.nist.gov/programs-projects/genome-bottle"
        },
        {
            "handoverType": {
                "id": "CUSTOM:000001",
                "label": "Project description"
            },
            "note": "Project description",
            "url": "https://www.nist.gov/programs-projects/genome-bottle"
        },
        {
            "handoverType": {
                "id": "CUSTOM:000001",
                "label": "Project description"
            },
            "note": "Project description",
            "url": "https://www.nist.gov/programs-projects/genome-bottle"
        }
    ],
    "meta": {
        "apiVersion": "v2.0.0",
        "beaconId": "org.ega-archive.beacon-ri-demo",
        "receivedRequestSummary": {
            "apiVersion": "v2.0.0",
            "filters": [],
            "includeResultsetResponses": "HIT",
            "pagination": {
                "limit": 10,
                "skip": 0
            },
            "requestParameters": {},
            "requestedGranularity": "record",
            "requestedSchemas": [],
            "testMode": false
        },
        "returnedGranularity": "record",
        "returnedSchemas": [
            {
                "entityType": "genomicVariation",
                "schema": "beacon-g_variant-v2.0.0"
            }
        ]
    },
    "response": {
        "resultSets": [
            {
                "exists": true,
                "id": "CINECA_synthetic_cohort_EUROPE_UK1",
                "results": [
                    {
                        "_id": "673b5a8d67777f76f8228074",
                        "caseLevelData": [
                            {
                                "biosampleId": "HG03771",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            }
                        ],
                        "identifiers": {
                            "genomicHGVSId": "NC_000022.11:g.16050075A>G"
                        },
                        "molecularAttributes": {
                            "aminoacidChanges": [
                                "."
                            ],
                            "geneIds": [
                                "CHR_START-DUXAP8"
                            ],
                            "molecularEffects": [
                                {
                                    "id": "ENSGLOSSARY:0000174",
                                    "label": "intergenic_region"
                                }
                            ]
                        },
                        "variantInternalId": "da5a95e4-bc26-11ee-b6b0-0242ac170002:A:G",
                        "variation": {
                            "alternateBases": "A",
                            "location": {
                                "interval": {
                                    "end": {
                                        "type": "Number",
                                        "value": 16050075
                                    },
                                    "start": {
                                        "type": "Number",
                                        "value": 16050074
                                    },
                                    "type": "SequenceInterval"
                                },
                                "sequence_id": "HGVSid:22:g.16050075A>G",
                                "type": "SequenceLocation"
                            },
                            "referenceBases": "G",
                            "variantType": "SNP"
                        }
                    },
                    {
                        "_id": "673b5a8d67777f76f8228075",
                        "caseLevelData": [
                            {
                                "biosampleId": "HG01932",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            }
                        ],
                        "identifiers": {
                            "genomicHGVSId": "NC_000022.11:g.16050655G>A"
                        },
                        "molecularAttributes": {
                            "aminoacidChanges": [
                                "."
                            ],
                            "geneIds": [
                                "CHR_START-DUXAP8"
                            ],
                            "molecularEffects": [
                                {
                                    "id": "ENSGLOSSARY:0000174",
                                    "label": "intergenic_region"
                                }
                            ]
                        },
                        "variantInternalId": "da67dd62-bc26-11ee-b6b0-0242ac170002:G:A",
                        "variation": {
                            "alternateBases": "G",
                            "location": {
                                "interval": {
                                    "end": {
                                        "type": "Number",
                                        "value": 16050655
                                    },
                                    "start": {
                                        "type": "Number",
                                        "value": 16050654
                                    },
                                    "type": "SequenceInterval"
                                },
                                "sequence_id": "HGVSid:22:g.16050655G>A",
                                "type": "SequenceLocation"
                            },
                            "referenceBases": "A",
                            "variantType": "SNP"
                        }
                    },
                    {
                        "_id": "673b5a8d67777f76f8228076",
                        "caseLevelData": [
                            {
                                "biosampleId": "HG01365",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG02337",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG02345",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG02501",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG02537",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG02577",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG02624",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG02878",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG03054",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG03060",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG03085",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG03363",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG03369",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG03394",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG03433",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG03476",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA18517",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA18520",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "NA18861",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA18876",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA18909",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "NA19028",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA19107",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "NA19129",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA19138",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA19214",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "NA19225",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "NA19463",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "NA19711",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA20126",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA20334",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            }
                        ],
                        "identifiers": {
                            "genomicHGVSId": "NC_000022.11:g.16050115G>A"
                        },
                        "molecularAttributes": {
                            "aminoacidChanges": [
                                "."
                            ],
                            "geneIds": [
                                "CHR_START-DUXAP8"
                            ],
                            "molecularEffects": [
                                {
                                    "id": "ENSGLOSSARY:0000174",
                                    "label": "intergenic_region"
                                }
                            ]
                        },
                        "variantInternalId": "da5cbf36-bc26-11ee-b6b0-0242ac170002:G:A",
                        "variation": {
                            "alternateBases": "G",
                            "location": {
                                "interval": {
                                    "end": {
                                        "type": "Number",
                                        "value": 16050115
                                    },
                                    "start": {
                                        "type": "Number",
                                        "value": 16050114
                                    },
                                    "type": "SequenceInterval"
                                },
                                "sequence_id": "HGVSid:22:g.16050115G>A",
                                "type": "SequenceLocation"
                            },
                            "referenceBases": "A",
                            "variantType": "SNP"
                        }
                    },
                    {
                        "_id": "673b5a8d67777f76f8228077",
                        "caseLevelData": [
                            {
                                "biosampleId": "HG02481",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG02485",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            }
                        ],
                        "identifiers": {
                            "genomicHGVSId": "NC_000022.11:g.16050954G>A"
                        },
                        "molecularAttributes": {
                            "aminoacidChanges": [
                                "."
                            ],
                            "geneIds": [
                                "CHR_START-DUXAP8"
                            ],
                            "molecularEffects": [
                                {
                                    "id": "ENSGLOSSARY:0000174",
                                    "label": "intergenic_region"
                                }
                            ]
                        },
                        "variantInternalId": "da7524a4-bc26-11ee-b6b0-0242ac170002:G:A",
                        "variation": {
                            "alternateBases": "G",
                            "location": {
                                "interval": {
                                    "end": {
                                        "type": "Number",
                                        "value": 16050954
                                    },
                                    "start": {
                                        "type": "Number",
                                        "value": 16050953
                                    },
                                    "type": "SequenceInterval"
                                },
                                "sequence_id": "HGVSid:22:g.16050954G>A",
                                "type": "SequenceLocation"
                            },
                            "referenceBases": "A",
                            "variantType": "SNP"
                        }
                    },
                    {
                        "_id": "673b5a8d67777f76f8228078",
                        "caseLevelData": [
                            {
                                "biosampleId": "HG01440",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            }
                        ],
                        "identifiers": {
                            "genomicHGVSId": "NC_000022.11:g.16050319C>T"
                        },
                        "molecularAttributes": {
                            "aminoacidChanges": [
                                "."
                            ],
                            "geneIds": [
                                "CHR_START-DUXAP8"
                            ],
                            "molecularEffects": [
                                {
                                    "id": "ENSGLOSSARY:0000174",
                                    "label": "intergenic_region"
                                }
                            ]
                        },
                        "variantInternalId": "da5f1c5e-bc26-11ee-b6b0-0242ac170002:C:T",
                        "variation": {
                            "alternateBases": "C",
                            "location": {
                                "interval": {
                                    "end": {
                                        "type": "Number",
                                        "value": 16050319
                                    },
                                    "start": {
                                        "type": "Number",
                                        "value": 16050318
                                    },
                                    "type": "SequenceInterval"
                                },
                                "sequence_id": "HGVSid:22:g.16050319C>T",
                                "type": "SequenceLocation"
                            },
                            "referenceBases": "T",
                            "variantType": "SNP"
                        }
                    },
                    {
                        "_id": "673b5a8d67777f76f8228079",
                        "caseLevelData": [
                            {
                                "biosampleId": "HG04014",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            }
                        ],
                        "identifiers": {
                            "genomicHGVSId": "NC_000022.11:g.16050972G>A"
                        },
                        "molecularAttributes": {
                            "aminoacidChanges": [
                                "."
                            ],
                            "geneIds": [
                                "CHR_START-DUXAP8"
                            ],
                            "molecularEffects": [
                                {
                                    "id": "ENSGLOSSARY:0000174",
                                    "label": "intergenic_region"
                                }
                            ]
                        },
                        "variantInternalId": "da769e38-bc26-11ee-b6b0-0242ac170002:G:A",
                        "variation": {
                            "alternateBases": "G",
                            "location": {
                                "interval": {
                                    "end": {
                                        "type": "Number",
                                        "value": 16050972
                                    },
                                    "start": {
                                        "type": "Number",
                                        "value": 16050971
                                    },
                                    "type": "SequenceInterval"
                                },
                                "sequence_id": "HGVSid:22:g.16050972G>A",
                                "type": "SequenceLocation"
                            },
                            "referenceBases": "A",
                            "variantType": "SNP"
                        }
                    },
                    {
                        "_id": "673b5a8d67777f76f822807a",
                        "caseLevelData": [
                            {
                                "biosampleId": "NA07357",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            }
                        ],
                        "identifiers": {
                            "genomicHGVSId": "NC_000022.11:g.16050958A>T"
                        },
                        "molecularAttributes": {
                            "aminoacidChanges": [
                                "."
                            ],
                            "geneIds": [
                                "CHR_START-DUXAP8"
                            ],
                            "molecularEffects": [
                                {
                                    "id": "ENSGLOSSARY:0000174",
                                    "label": "intergenic_region"
                                }
                            ]
                        },
                        "variantInternalId": "da75df02-bc26-11ee-b6b0-0242ac170002:A:T",
                        "variation": {
                            "alternateBases": "A",
                            "location": {
                                "interval": {
                                    "end": {
                                        "type": "Number",
                                        "value": 16050958
                                    },
                                    "start": {
                                        "type": "Number",
                                        "value": 16050957
                                    },
                                    "type": "SequenceInterval"
                                },
                                "sequence_id": "HGVSid:22:g.16050958A>T",
                                "type": "SequenceLocation"
                            },
                            "referenceBases": "T",
                            "variantType": "SNP"
                        }
                    },
                    {
                        "_id": "673b5a8d67777f76f822807b",
                        "caseLevelData": [
                            {
                                "biosampleId": "HG01438",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG01670",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG01912",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG02111",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG02113",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG02583",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG02611",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG02648",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG02676",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG02716",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG02774",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG02805",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG02808",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG02811",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG02817",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG02854",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG02881",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG02890",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG02943",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG03049",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG03063",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG03163",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG03202",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG03304",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG03518",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "HG03538",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA18874",
                                "zygosity": {
                                    "id": "GENO:GENO_0000136",
                                    "label": "1/1"
                                }
                            },
                            {
                                "biosampleId": "NA19025",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA19095",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "NA19116",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "NA19172",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "NA19248",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "NA19257",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "NA19317",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA19404",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA19428",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            },
                            {
                                "biosampleId": "NA20355",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            }
                        ],
                        "identifiers": {
                            "genomicHGVSId": "NC_000022.11:g.16050213C>T"
                        },
                        "molecularAttributes": {
                            "aminoacidChanges": [
                                "."
                            ],
                            "geneIds": [
                                "CHR_START-DUXAP8"
                            ],
                            "molecularEffects": [
                                {
                                    "id": "ENSGLOSSARY:0000174",
                                    "label": "intergenic_region"
                                }
                            ]
                        },
                        "variantInternalId": "da5e0d32-bc26-11ee-b6b0-0242ac170002:C:T",
                        "variation": {
                            "alternateBases": "C",
                            "location": {
                                "interval": {
                                    "end": {
                                        "type": "Number",
                                        "value": 16050213
                                    },
                                    "start": {
                                        "type": "Number",
                                        "value": 16050212
                                    },
                                    "type": "SequenceInterval"
                                },
                                "sequence_id": "HGVSid:22:g.16050213C>T",
                                "type": "SequenceLocation"
                            },
                            "referenceBases": "T",
                            "variantType": "SNP"
                        }
                    },
                    {
                        "_id": "673b5a8d67777f76f822807c",
                        "caseLevelData": [
                            {
                                "biosampleId": "HG01914",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG01967",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "1/0"
                                }
                            },
                            {
                                "biosampleId": "HG03103",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            }
                        ],
                        "identifiers": {
                            "genomicHGVSId": "NC_000022.11:g.16050994G>C"
                        },
                        "molecularAttributes": {
                            "aminoacidChanges": [
                                "."
                            ],
                            "geneIds": [
                                "CHR_START-DUXAP8"
                            ],
                            "molecularEffects": [
                                {
                                    "id": "ENSGLOSSARY:0000174",
                                    "label": "intergenic_region"
                                }
                            ]
                        },
                        "variantInternalId": "da781bf0-bc26-11ee-b6b0-0242ac170002:G:C",
                        "variation": {
                            "alternateBases": "G",
                            "location": {
                                "interval": {
                                    "end": {
                                        "type": "Number",
                                        "value": 16050994
                                    },
                                    "start": {
                                        "type": "Number",
                                        "value": 16050993
                                    },
                                    "type": "SequenceInterval"
                                },
                                "sequence_id": "HGVSid:22:g.16050994G>C",
                                "type": "SequenceLocation"
                            },
                            "referenceBases": "C",
                            "variantType": "SNP"
                        }
                    },
                    {
                        "_id": "673b5a8d67777f76f822807d",
                        "caseLevelData": [
                            {
                                "biosampleId": "NA20321",
                                "zygosity": {
                                    "id": "GENO:GENO_0000458",
                                    "label": "0/1"
                                }
                            }
                        ],
                        "identifiers": {
                            "genomicHGVSId": "NC_000022.11:g.16050996T>C"
                        },
                        "molecularAttributes": {
                            "aminoacidChanges": [
                                "."
                            ],
                            "geneIds": [
                                "CHR_START-DUXAP8"
                            ],
                            "molecularEffects": [
                                {
                                    "id": "ENSGLOSSARY:0000174",
                                    "label": "intergenic_region"
                                }
                            ]
                        },
                        "variantInternalId": "da7a6fc2-bc26-11ee-b6b0-0242ac170002:T:C",
                        "variation": {
                            "alternateBases": "T",
                            "location": {
                                "interval": {
                                    "end": {
                                        "type": "Number",
                                        "value": 16050996
                                    },
                                    "start": {
                                        "type": "Number",
                                        "value": 16050995
                                    },
                                    "type": "SequenceInterval"
                                },
                                "sequence_id": "HGVSid:22:g.16050996T>C",
                                "type": "SequenceLocation"
                            },
                            "referenceBases": "C",
                            "variantType": "SNP"
                        }
                    }
                ],
                "resultsCount": 6654,
                "resultsHandover": {
                    "handoverType": {
                        "id": "CUSTOM:000001",
                        "label": "Project description"
                    },
                    "note": "Project description",
                    "url": "https://www.nist.gov/programs-projects/genome-bottle"
                },
                "setType": "dataset"
            }
        ]
    },
    "responseSummary": {
        "exists": true,
        "numTotalResults": 13308
    }
}
`
GET with filter does not work
`(master)> http GET http://localhost:5050/api/g_variants?referenceName=22
HTTP/1.1 400 Bad Request
Content-Length: 228
Content-Type: application/json; charset=utf-8
Date: Mon, 18 Nov 2024 15:25:34 GMT
Server: Python/3.10 aiohttp/3.8.1
{
    "error": {
        "errorCode": "400",
        "errorMessage": "Bad Request"
    },
    "meta": {
        "apiVersion": "v2.0.0",
        "beaconId": "org.ega-archive.beacon-ri-demo",
        "receivedRequestSummary": "",
        "returnedGranularity": "record",
        "returnedSchemas": []
    }
}`

This same command works fine with the official May release.

So I can't say if the issue is fixed with the "referenceName" filter ...

**Comment by costero-e** [comment]
Hi @albodrug,
thanks for the feedback. The explanation is request parameters can't be thrown on their own alone. Following documentation for [Beacon v2](https://docs.genomebeacons.org/variant-queries/), you can throw a query that includes referenceName but not referenceName alone. For example, for a _Sequence Query_, you could throw referenceName, start, alternateBases, referenceBases and assemblyId together. If one of those is missing, you will get a 400 bad request response.
Tell me if this has worked for you.
Thanks,
Oriol

**Comment by albodrug** [comment]
Hello Oriol, 

I tried the latest commit version and the issue seems to be fixed!

Indeed I can no longer send requests with referenceName only. Also another difference is that previously in the POST or GET requests I did not have to specify the assemblyId. But now if I don't specify it, I am getting a 400 bad request response, so assemblyId seems to be necessary too now.

 >request parameters can't be thrown on their own alone

Ok, I see. In the May release "referenceName" can be used alone (GET and POST), that is why I was confused. It would have been useful for me but it makes sense not to allow it. I could also use "start" and "end" alone, without specifying the chromosome name, and that was quite confusing because I was getting the variants in the specified interval for all the chromosomes. I could also specify only the variantType. I will stop using the May release then, because it caused a lot of confusion on my side. 

Thank you a lot!
Alex

---
Posting some remarks in case it can help someone:

1) I had another issue, that is if the transformed my VCF (GCRh38) using ri-tools-v2 and the VCF had chromosome names written as "chr1" and not as "1" I could not find anything through the API using
` "referenceName" : "chr1"` or `"referenceName" : 1`
It looks like this is caused by the chromosome name being used to reconstruct the "genomicHGVSId" sequence identifiers ("NC00001" "1" being the chromosome id, and become "NC0000chr1" in my case). I renamed everything as "1" instead of "chr1" and this fixed it. 
2) To easily get the chromosome name I also slightly modified the ri-tools-v2, so the genomicVariants.json BFF displays "chr" : "1" in the variant.location portion. 
3) The GET request that works for me now
`http GET http://localhost:5050/api/g_variants?assemblyId=GRCh38\&referenceName=2\&start=112829751\&end=112836779`
4) The POST request that works for me now, assemblyId and referenceName are mandatory now (compared to previous versions)
`curl -X POST http://localhost:5050/api/g_variants -H "Content-Type: application/json" -d '{
  "meta": {
    "apiVersion": "2.0"
  },
  "query": {
    "requestParameters": {
    	"referenceName" : "2",
	"start": [ 112829751 ],
        "end": [ 112836779 ],
        "assemblyId" : "GRCh38"
    },
    "filters": [],
    "includeResultsetResponses": "HIT",
    "pagination": {
      "skip": 0,
      "limit": 1000
    },
    "testMode": false,
    "requestedGranularity": "record"
  }
}'`
